workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -111,14 +111,16 @@ def _copy_data(archive_root, config, to_archive=True):
     # where first path is path in manager, second is path in snapshot.
     # If paths are relative then should be relative to file server (path
     # in manager) and snapshot archive (path in snapshot). If paths are
-    # absolute then should point to proper data in manager/snapshot archive
+    # absolute then should point to proper data in manager/snapshot archive.
+    # The last element denotes whether the directory should be overwritten.
     data_to_copy = [
-        (config.file_server_blueprints_folder, 'blueprints'),
-        (config.file_server_uploaded_blueprints_folder, 'uploaded-blueprints'),
-        (config.file_server_uploaded_plugins_folder, 'plugins')
+        (config.file_server_blueprints_folder, 'blueprints', True),
+        (config.file_server_uploaded_blueprints_folder, 'uploaded-blueprints',
+         True),
+        (config.file_server_uploaded_plugins_folder, 'plugins', False)
```

Comment:
Moreover, we should probably use:

''' python
os.system("cp -r src/* target")

Because this shutil thing explodes if you want to copy a directory onto an existing one. This will result in a failure when restoring a snapshot with plugins on a manager on which we have just created this snapshot..